contracts/contracts/coordination/IFeeModel.sol

```diff
@@ -9,17 +9,35 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  * @notice IFeeModel
  */
 interface IFeeModel {
-    function currency() external view returns (IERC20);
```

Comment:
Why removing the `currency()` interface here? Maybe it's because we're moving towards a credits system? In any case, functions to calculate cost are still defined with a reference currency token, so I'd keep this.

Reply:
because FeeModel is more flexible, just imagine fee model without payment